files/zh-cn/web/javascript/reference/global_objects/string/index.md

```diff
@@ -92,48 +92,48 @@ areEqual("ß", "ss", "de"); // false
 areEqual("ı", "I", "tr"); // true
 ```
 
-`localeCompare()` 方法以与 `strcmp()` 类似的方式启用字符串比较——它允许以区域感知的方式进行对字符串进行排序。
+`localeCompare()` 方法使字符串比较可以像 `strcmp()` 函数一样进行——它允许以本地化感知的方式对字符串进行排序。
 
 ### 字符串原始值和字符串对象
 
-注意，JavaScript 区分 `String` 对象和{{Glossary("Primitive", "原始字符串")}}值（{{jsxref("Boolean")}} 和 {{jsxref("Global_Objects/Number", "Number")}} 也是如此）。
+注意，JavaScript 区分 `String` 对象和{{Glossary("Primitive", "原始字符串", "", 1)}}值（{{jsxref("Boolean")}} 和 {{jsxref("Global_Objects/Number", "Number")}} 也是如此）。
 
-字符串字面量（使用单引号或者双引号表示）和从非构造函数上下文中的 `String` 调用返回的字符串（即在不使用 {{jsxref("Operators/new", "new")}} 关键字的情况下调用）是原始字符串。在要对原始字符串调用方法或者发生属性查找的上下文中，JavaScript 将自动的包装原始字符串并调用方法或在包装对象上执行属性查找。
+字符串字面量（使用单引号或者双引号表示）和从非构造函数上下文中的 `String` 调用返回的字符串（即在不使用 {{jsxref("Operators/new", "new")}} 关键字的情况下调用）是原始字符串。在需要调用原始字符串的方法或进行属性查找的上下文中，JavaScript 将自动的包装原始字符串并在包装对象上调用方法或执行属性查找。
 
 ```js
-const strPrim = "foo"; // A literal is a string primitive
-const strPrim2 = String(1); // Coerced into the string primitive "1"
-const strPrim3 = String(true); // Coerced into the string primitive "true"
-const strObj = new String(strPrim); // String with new returns a string wrapper object.
+const strPrim = "foo"; // 字面量是一个字符串原始值
+const strPrim2 = String(1); // 被强制转换为字符串原始值“1”
+const strPrim3 = String(true); // 被强制转换为字符串原始值“true”
+const strObj = new String(strPrim); // 使用 `new` 关键字调用 `String` 构造函数返回一个字符串包装对象。
 
 console.log(typeof strPrim); // "string"
 console.log(typeof strPrim2); // "string"
 console.log(typeof strPrim3); // "string"
 console.log(typeof strObj); // "object"
 ```
 
-> **警告：** 你应该发现你自己很少使用 `String` 构造函数。
+> **警告：** 你应该发现你自己很少使用 `String` 作为构造函数。
```

Comment:
actually, I like the previous one /:

Reply:
`String()` called without `new` acts like `toString()`, and can be used for converting something into a String primitive value. Considering this case, `String` is a common method. The "as" in the original sentence should not be omitted in the translation.

Reply:
/cc @jasonren0403

Reply:
refer https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/string/string/index.md#return-value

Reply:
I knew the original text is "using `String` as a constructor". But we usually say "将...作为", I rarely hear "使用...作为".

Reply:
Found other instances similar to this

https://github.com/mdn/translated-content/blob/1455c79e0b644d74c669494a47d3cda11c746ee7/files/zh-cn/web/javascript/reference/global_objects/number/number/index.md?plain=1#L30

https://github.com/mdn/translated-content/blob/1455c79e0b644d74c669494a47d3cda11c746ee7/files/zh-cn/web/javascript/reference/global_objects/boolean/boolean/index.md?plain=1#L32

Maybe we should just align with them

Reply:
> I knew the original text is "using `String` as a constructor". But we usually say "将...作为", I rarely hear "使用...作为".

If using "将...作为", here should be translated into "将 `String` 作为构造函数进行使用".
I don't think it will be more proper than using "使用 ... 作为 ... " as the translation of "using ... as ..."

Reply:
I would translate "call X as a constructor" as 「将 X 作为构造函数使用」, but in this case I would make the voice a bit more assertive:  「你应该基本不会将 `String` 作为构造函数使用」.